### PR TITLE
Feature: Add Unique HTML IDs to .dropzones, #fileinputs

### PR DESCRIPTION
**Pull request by noi5e**
See issues #8926 #8670 #8478

Adds unique HTML IDs to .dropzones and #fileinputs in comment forms.

Edit Comment

- **A**: _currently_: `.dropzone` (with no ID) wrapping `#fileinput` (which is used twice in the same comment form, or multiple times per page)
  - _after merging_: `#dropzone-small-edit-22`
  - _after merging_: `#fileinput-button-edit-22`
  - _after merging_: `#dropzone-reply-22` for reply forms, `#dropzone-main` for base comment form in notes, questions, and wikis
- **B**: _currently_: `.dropzone` (with no ID) wrapping the form's entire face, same class as A
  - _after merging_: `#dropzone-large-reply-20` in reply forms and `#dropzone-large-main` in main forms
- **C**: _currently_: second `#fileinput` here (distinct from A, but with the same ID)
  - this is now `#fileinput-choose-one-reply-1234` or `#fileinput-choose-one-main`

As usual, had to rewrite some of the old system tests that relied on the old classes and IDs. 

There's probably more tests to rewrite before this is ready-to-merge, thought it would be faster/easier to see in CI rather than run tests locally.

---
(This issue is part of the larger Comment Editor Overhaul Project with Outreachy. Refer to Planning Issue #8775 for more context)

### CODE REVIEW

File: app/views/comments/_edit.html.erb
Comment (jywarren) [inline app/views/comments/_edit.html.erb:11]: here you could use an HTML comment perhaps? `<!-- comment -->` just to simplify a little?

File: app/views/comments/_form.html.erb
Comment (jywarren) [inline app/views/comments/_form.html.erb:75]: Will this change styling at all? Noting that sometimes adding unique classnames instead of ids can be better since elements are allowed to have more than one class but only one ID. I tend to use classes instead of IDs for that reason, but it's just a convention or habit, so no big deal here, just curious!

File: app/views/notes/_comment.html.erb
Comment (jywarren) [inline app/views/notes/_comment.html.erb:169]: so much more readable, thank you!

File: app/views/comments/_form.html.erb
Comment (jywarren) [inline app/views/comments/_form.html.erb:35]: There's a case possibly to be made that logic doesn't belong in a template, but i see why this may be an exception. But just to explore the possibility, would it be possible to wrap this logic in a function and do it in a single line in the template, with the logic tucked into the `application_helper.rb` (helpers is where template-specific logic often goes, by convention)?

But, i'm not feeling strongly either way, there are pros and cons - it might be more readable in pure Ruby in the helper, but it's also more work to go and find it...

**Review by jywarren — APPROVED** [review]
This looks amazing @noi5e -- i think it's pretty much good to merge unless you want to do more; my questions below are clarifications and "just checking" so i defer to you on a final call on any of them. If it's ready to merge just note that in a comment and we can do it! 

Fantastic work here!!! 👍

File: app/views/comments/_form.html.erb
Comment (noi5e) [reply app/views/comments/_form.html.erb:75]: Oh, that's a really good point about styling. I did a simple search in VSCode for `#fileinput` and `input#fileinput` but it didn't come up with anything in `/assets/stylesheets`, so I think this is good. It looks as normal when I load it up in development too.

That's also a good point about classes vs. IDs. Since my ultimate goal is to be able to use `e.target` to set `$D.selected` upon file upload, I thought an ID made the most sense here... Good reminder though that I can use classes in other situations.

I was aware of the one ID per page rule. `dropzone_large_id` includes the comment ID, so it's going to resolve to a unique ID every time this form is rendered (`#fileinput-reply-large-22` or something like that)... Which means that there shouldn't be any conflicting IDs here, at least throughout `comments-list`!

File: app/views/comments/_form.html.erb
Comment (noi5e) [reply app/views/comments/_form.html.erb:35]: Thank you! This is great feedback, and helps me learn more about Rails conventions (which I'm going to be learning for a while). I went ahead and put the logic in `application_helper.rb`, just for the sake of developing good habits.

**Comment by noi5e** [comment]
I just pushed the commits, so this one's ready to merge!

**Comment by jywarren** [comment]
Awesome would you like to take code climates suggestions or just resolve them? Thanks!!!

**Comment by noi5e** [comment]
Just pushed the commits, it's ready (again) now!

**Comment by jywarren** [comment]
Hooray 🎉👏👏

**Pull request merged**
